itez/beneficiary/models.py

```diff
@@ -343,3 +344,156 @@ class WorkDetail(models.Model):
+        choices=IP_TYPES,
+        default=NGO,
+    )
+
+    is_active = models.BooleanField(
+        _('Is Active'),
+        default=True,
+        help_text=_('Is still an active Implementing Partner'),
+    )
+
+    def __str__(self):
+        return self.name
+
+class FacilityType(models.Model):
+    name = models.CharField(
+        _('Facility Type Name'),
+        max_length=200,
+    )
+
+    def __str__(self):
+        return self.name.title()
+
+
+class Facility(models.Model):
+    hmis_code = models.CharField(
+        _('HMIS Code'),
+        max_length=100,
+        null=True,
+        blank=True
+    )
+    province = models.ForeignKey(
+        Province,
+        default="",
+        on_delete=models.CASCADE,
+        verbose_name=_("Province"),
+    )
+
+    district = models.ForeignKey(
+        District,
+        default=1,
+        on_delete=models.CASCADE,
+        help_text=_("District in which the facility is located"),
+        max_length=250,
+    )
+
+    name = models.CharField(
+        _('Name'),
+        max_length=200,
+        help_text=_(
+            "Just enter name without 'Hostpial' or 'Clinic`, i.e for `Kitwe General Hospital` just enter `Kitwe General`.")
+    )
+
+    facility_type = models.ForeignKey(
+        FacilityType,
+        default=1,
+        on_delete=models.CASCADE,
+        help_text=_("Facility Type, i.e 'Hospital, Clinic etc"),
+        max_length=250,
+    )
+
+    implementing_partner = models.ForeignKey(
+        ImplementingPartner,
+        on_delete=models.SET_NULL,
+        help_text=_("Related Implementing Partner."),
+        max_length=250,
+        null=True,
+        blank=True
+    )
+
+    class Meta:
+        verbose_name = 'Facility'
+        verbose_name_plural = 'Facilities'
+
+    def save(self):
+        if self.facility_type.name.lower() in self.name.lower():
+            self.name = (self.name).title()
+        else:
+            f"{self.name.title()} {self.facility_type.name.title()}"
+        super(Facility, self).save()
+
+    def __str__(self):
+        return str(self.name)
+
+class ServiceProvider(models.Model):
+    QUALIFICATIONS = (
+        ("certificate", _("Certificate")),
+        ("diploma", _("Diploma")),
+        ("degree", _("Degree")),
+        ("masters", _("Masters")),
+        ("doctrate", _("Doctrate")),
+        ("phd", _("PHD")),
+    )
+
+    first_name = models.CharField(
+        _("First Name"),
+        max_length=200
+    )
+    last_name = models.CharField(
+        _("Last Name"),
+        max_length=200
+    )
+    date_of_birth = models.DateField(
+        _("Date of Birth"),
+        null=True,
+        blank=True
+    )
+    department = models.CharField(
+        _("Department"),
+        max_length=200,
+        null=True,
+        blank=True
+    )
+    facility = models.ForeignKey(
+        Facility,
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL
+    )
+    qualification = models.CharField(
```

Comment:
I.e `Degree` Degree in what?
lets capture the qualification Name itself too so that it makes sense to say `Degree in Nursing` Degree coming from qualification types and Nursing from the  name

Reply:
I created a separate model for qualifications.